waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -694,11 +703,12 @@ proc createMembershipList*(n: int): RlnRelayResult[(
     let keyTuple = (keypair.idKey.inHex, keypair.idCommitment.inHex)
     output.add(keyTuple)
 
-    # insert the key to the Merkle tree
-    let inserted = rln.insertMember(keypair.idCommitment)
-    if not inserted:
-      return err("could not insert the key into the Merkle tree")
-
+    idCommitments.add(keypair.idCommitment)
+    
+  # Insert members into tree
+  let membersAdded = rln.insertMembers(0, idCommitments)
```

Comment:
Do we know what the is state of the tree when insertion fails for one of the id commitments in the list? will the prior insertions revert as well?

Reply:
Addressed in https://github.com/vacp2p/zerokit/pull/66

Reply:
Thanks, then it is worth explaining in the `insertMembers` wapper proc that the action is atomic, i.e., all or none.

Reply:
Addressed in 8c08ed16